database/engine/global_uuid_map/global_uuid_map.c

```diff
@@ -244,10 +244,12 @@ int find_or_generate_guid(void *object, uuid_t *uuid, GUID_TYPE object_type, int
         uv_rwlock_wrunlock(&global_lock);
         return rc;
```

Comment:
You could also add a `if (rc) freez(target_object);` here.

Reply:
True, I will add